File: solutions/python/grains/1/grains.py
def square(number):
    if number < 1 or number > 64:
        raise ValueError("Number value is out of range")

    return 2**(number - 1)


def total():
    grains = 0

    for square_number in range(1, 65):
        grains += square(square_number)
    
    return grains

File: solutions/ruby/grains/1/grains.rb
module Grains
  def self.square(square_number)
    raise ArgumentError if square_number < 1 || square_number > 64

    2**(square_number - 1)
  end

  def self.total
    total = 0
    (1..64).each { |position| total += square(position) }
    total
  end
end
